fence/agents/heuristics_ping/fence_heuristics_ping.py

```diff
@@ -0,0 +1,195 @@
+
+	# start a ping-process per target
+	for target in targets:
+		ping_cmd = "%s -n -q -W %d -c %d -i %d %s" % (
+			PING_BINARY, timeout, count, interval, target)
+		logging.info("Running command: %s", ping_cmd)
+		try:
+			p[target] = subprocess.Popen(shlex.split(ping_cmd),
+				stdout=subprocess.PIPE);
+		except OSError:
+			p[target] = None
+
+	# collect the results of the ping-processes
+	for target in targets:
+		good = 0
+		if p[target] != None:
+			p[target].wait()
+			if p[target].returncode == 0:
+				for line in p[target].stdout:
+					searchres = packet_count.search(line)
+					if searchres:
+						good = int(searchres.group(1))
+						break
+				if good >= good_required:
+					logging.info("ping target %s received %d of %d" \
+						% (target, good, count))
+					continue
+			failcount += 1
+			logging.info("ping target %s received %d of %d and thus failed"
+				% (target, good, count))
+		else:
+			failcount += 1
+			logging.error("ping target %s failed on OS level" % target)
+
+	if failcount > maxfail:
+		exitcode = False
+		# "off" was requested so fake "on" to provoke failure
+		get_power_fake.result = "on"
+	else:
+		get_power_fake.result = "off"
+
+	return exitcode
+
+
+def define_new_opts():
+	all_opt["ping_count"] = {
+		"getopt" : ":",
+		"longopt" : "ping-count",
+		"required" : "0",
+		"help" : "--ping-count=[number]          Number of ping-probes to send",
+		"shortdesc" : "The number ob ping-probes that is being sent per target",
+		"default" : "10",
+		"order" : 1
+		}
+
+	all_opt["ping_goodcnt"] = {
+		"getopt" : ":",
+		"longopt" : "ping-goodcnt",
+		"required" : "0",
+		"help" : "--ping-goodcnt=[number]        Number of positive ping-probes required",
+		"shortdesc" : "The number ob positive ping-probes required to account a target as available",
+		"default" : "8",
+		"order" : 1
+		}
+
+	all_opt["ping_interval"] = {
+		"getopt" : ":",
+		"longopt" : "ping-interval",
+		"required" : "0",
+		"help" : "--ping-interval=[seconds]      Seconds between ping-probes",
+		"shortdesc" : "The interval in seconds between ping-probes",
+		"default" : "1",
+		"order" : 1
+		}
+
+	all_opt["ping_timeout"] = {
+		"getopt" : ":",
+		"longopt" : "ping-timeout",
+		"required" : "0",
+		"help" : "--ping-timeout=[seconds]       Timeout for individual ping-probes",
+		"shortdesc" : "The timeout in seconds till an individual ping-probe is accounted as lost",
+		"default" : "2",
+		"order" : 1
+		}
+
+	all_opt["ping_maxfail"] = {
+		"getopt" : ":",
+		"longopt" : "ping-maxfail",
+		"required" : "0",
+		"help" : "--ping-maxfail=[number]        Number of failed ping-targets allowed",
+		"shortdesc" : "The number of failed ping-targets to still account as overall success",
+		"default" : "0",
+		"order" : 1
+		}
+
+	all_opt["ping_targets"] = {
+		"getopt" : ":",
+		"longopt" : "ping-targets",
+		"required" : "1",
+		"help" : "--ping-targets=tgt1:tgt2       Colon separated list of ping-targets",
```

Comment:
we are using comma for setting multiple values at once (e.g. --plug 1,2,3)

Reply:
this and going IPv6 using colons as separators isn't that wise either ...